File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        char[] num = Integer.toString(numberToCheck).toCharArray();
        int digits = num.length;
        int sum = 0;
        for (int i = 0; i<num.length; i++) {
            sum += Math.pow(Character.getNumericValue(num[i]),num.length);
        }
        return sum == numberToCheck;
        

    }

}

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include <stdio.h>
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
    if (candidate < 0) return false;
    if (candidate < 10) return true;
    int number = candidate;
    int suma = 0;
    int num_digitos = floor(log10(candidate)) + 1;
    
    while (number >= 10) {
        int digito = number %10;
        suma += pow(digito, num_digitos);
        number /= 10;
    }
    
    return suma == candidate;
}